src/shogun/io/TBOutputFormat.cpp

```diff
@@ -106,6 +107,13 @@ tensorflow::Event TBOutputFormat::convert_scalar(
 		    "Unsupported type %s", value.first.get_value().type_info().name());
 	}
 
+	auto write_summary = [&summaryValue=summaryValue, &value](auto type) {
+		summaryValue->set_simple_value(
+		    any_cast<decltype(type)>(value.first.get_value()));
+	};
+
+	sg_for_each_type(value.first.get_value(), sg_all_types, write_summary);
```

Comment:
since you have the vector thingi now in there, maybe also refactor the histogram bit?

Reply:
I was thinking about that, but it uses std::vector instead of SGVector? Is there some implicit casting going on?

Reply:
mmh phew ...
I would like to think it is possible to refactor this to SGVector?
On the other hand, it might be nice to support std:: structures as well

Reply:
I think the conversion to std::vector is necessary, because tensor board doesn't know about SGVector, like in this case:
https://github.com/shogun-toolbox/shogun/blob/2f0b9cfe59d18fec38904fbc74ba5aa55a531e33/src/shogun/io/TBOutputFormat.cpp#L125-L135